Add support for RedHat

michal800106 opened the pull request.

zielonkam committed:
Add support for RedHat

Change-Id: Ifad64c3fe8ed6f6a3e0a51b82a669d020033f147

zielonkam committed:
Add support for RedHat

Change-Id: Ifad64c3fe8ed6f6a3e0a51b82a669d020033f147

Commit:
Change specific version of monasca-agent to be installed

This is custom version without pylint requirement

Change-Id: Icb500b0ac016e9bfdc7e0044f369b26488d14368

Commit:
Remove virtualenv usage, because of download problems from other hosts

Commit:
Revert "Remove virtualenv usage, because of download problems from other hosts"

This reverts commit f010c7ebc59e2ad21393b5240a510ee74cdb68b5.

mattibf committed:
Check if pip.conf already exists and skip if true

Commit:
Upgrade monasca-agent version: 1.0.31-dev => 1.1.11

Commit:
Introduce a separate virtualenv directory

Change-Id: I537fe762da3cc4f6729a68471bf87c3e7ba9c03d

Commit:
Update monasca-agent version: 1.1.12

Commit:
Merge branch 'Introduce separate virtualenv directory'

Commit:
Fix bug 11821: change group and permissions of monasca-agent plugin config files

Commit:
Fix bug 11821: change group and permissions of monasca-agent plugin config files

Commit:
Refactor creating configuration

Because monasca-setup handler was run at the end of a playbook, it
was overwriting custom 'process' plugin configuration. We had to make
ansible assemble configuration from default and provided custom
one. This solution is generic and will work also in case of other
plugins.

Change-Id: I590ba36e618ae82465e10bcdce6d28fb6262bcf5

Commit:
Revert "Fix bug 11821: change group and permissions of monasca-agent plugin config files"

This reverts commit 73db978702553a5f6efd33f63f3adef20a4521bc, reversing
changes made to 3ea79484f9875b380f7ba70c8bb8aaeb1ca7cb74.

Commit:
Add installing psycopg2 for postgres checks

Installation is only fired when there's postgres configuration in
monasca_checks defined.

Change-Id: I62e5c3bd38d526db0bb95295db7e43ee06c920ae

Commit:
Add configurable option for logs directory

Change-Id: Iab4a509a68d6d4fca37244edb18cb4b21985bae2

Commit:
Fix typo

Change-Id: Iddebdedb41fcb41d68265727d44156e5d4fcc647

Commit:
Use more accurate variable for postgresql bin dir

Change-Id: If35efa1f079fddbd609071b44d692d5c731c1ba1

zielonkam committed:
Merge remote-tracking branch 'github_fj/master' into gitlab_master

tkuhlman commented on tasks/configure.yml:
```diff
@@ -8,17 +16,48 @@
-- name: Create reconfigure script
-  template: dest="{{agent_reconfigure_script}}" src=monasca-reconfigure.j2 owner=root group=root mode=750
-  notify: run monasca-setup
+- name: Create additional plugins config
```
There is some rather complicated reworking of the additional configuration to fix things up. Looking at it a bit the only problem seems to be that the '--overwrite' flag is specified in the monasca-reconfigure script and it shouldn't be in the case of any additional configuration being specified. Wouldn't fixing that solve the problems with additional config and leave the merging logic to monasca-setup?

tkuhlman commented on defaults/main.yml:
```diff
@@ -4,8 +4,16 @@
+monasca_agent_log_dir: "{{ monasca_log_dir }}/agent"
 
 skip_install: False
-monasca_virtualenv_dir: /opt/monasca
```
Multiple Monasca roles reuse this same venv so lets not change the name.

witekest replied:
The new name gives more flexibility. You may want to install agent in the separate virtualenv from the rest of monasca components.

tkuhlman commented on defaults/main.yml:
```diff
@@ -4,8 +4,16 @@
 monasca_checks: {}
 monasca_agent_system_only: False
 agent_reconfigure_script: /usr/local/bin/monasca-reconfigure
+monasca_agent_check_frequency: 60
```
I prefer to rely on the default check frequency built into the agent itself rather than setting one here.

witekest replied:
Done.

Commit:
Add flag for overwriting configuration

Custom configuration won't be overwritten if you set
`monasca_agent_overwrite_config` to True.

Change-Id: I30391e73ce08db7ab08043fe221441eaf1f1c6e4

Commit:
Add appropriate acess right during monasca-agent installation

Change-Id: Ifc3e39763ad2563c7673a6ade7a6c6578268ec50

Commit:
Add additional group

Change-Id: I30b8c82f1066190e8d566d56088dd70eed643d54

Commit:
Setup groups mon-agent and monasca

Change-Id: I640fac670e46820046c821534b65456c0eacde65

zawisz committed:
Fixed permissions and cleaned up agent config files

Remove config file fragments, prevent backup generation.
Moved the task which ensures that proper permissions are set to the
bottom so that all files generated by handlers have proper permissions.

Change-Id: Iad96ad35e58813e22a660dafaadc98ca1dd56fb8

zielonkam committed:
Adjusting monasca-agent role

- add environment variable for pip task

Change-Id: If64a4412f58e5395ac163890ed681a65a109eb7b

Commit:
Merge remote-tracking branch 'hp/master'

zawisz committed:
Create directory for logs with correct attributes

This prevents the problem with the process not starting if it can't
write to log files. The situation may occur if the agent is removed
along with the system user and then reinstalled.

Change-Id: I6a3dd48103829e3f1e0ab5841be1dd5a93446bbf

zawisz committed:
Install psycopg2 only if PostgreSQL binaries exist

Change-Id: Ib29cac4e7dbab8049d6a1ee00a3f8916defdb6a1

Commit:
Changes for Monasca Mitaka

Added installation of extra packages that are required for
monasca-agent.

Change-Id: I07777044eaffca8e5d1b9107ac6af1d5b8112c8c

haruki-yamanashi committed:
Remove postgresql adapter

Remove installing pyscopg2 from task and postgresql_dir from main.yml. 

Change-Id: I47cf45f607fb0856bc1b7c66a2a90ae5ff3f0d86

haruki-yamanashi committed:
Change for installation pbr to virtual environment

This patch installs pbr into virtual environment
to deal with user proxy setting.

Change-Id: I0d17ba56ad8b29d88302f7934ee1e5a304a7072c

zawisz committed:
Flush handlers before file attributes are set

The purpuse of this flush is to trigger monasca-setup
before the role sets proper attributes of files.
Monasca-setup manipulates configuration files and
may change their owner and group to root.

Change-Id: I5a4fae31b9ce0e7510a220f0041c6ddfded643b0
